### PR TITLE
Misc cleanup of graphics api from private branch

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/graphics/src/test/wscript [hunk 1]
```diff
@@ -31,10 +31,17 @@ def build(bld):
...
-                    use = 'TESTMAIN APP DDF DLIB PROFILE_NULL'.split() + graphics_libs + platform_lib + glfw_lib + extra_libs,
-                    web_libs = ['library_sys.js', glfw_js],
-                    target = 'test_app_graphics')
+        if platform_supports_feature(bld.env.PLATFORM, 'opengl', {}):
+            extra_symbols += ['GraphicsAdapterOpenGL']
+
+        if bld.env.PLATFORM in ('arm64-nx64'):
+            glfw_lib = []
+
+        if len(extra_symbols) > 0:
```
Comment (JCash) [inline engine/graphics/src/test/wscript:40]: If no backend is selected, we cannot build this example.
TODO: Make sure this app works for all platforms

File: engine/graphics/src/vulkan/graphics_native.cpp [hunk 2]
```diff
@@ -146,33 +146,6 @@ namespace dmGraphics
         }
     }
 
-    uint32_t VulkanGetDisplayDpi(HContext context)
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_native.cpp:149]: Some old code that was left behind

File: engine/graphics/src/vulkan/graphics_vulkan.cpp [hunk 3]
```diff
@@ -1128,6 +1128,15 @@ namespace dmGraphics
...
     static bool VulkanIsSupported()
     {
+#if defined(DM_PLATFORM_VENDOR)
+        // If we are on a private platform, and this driver is registered, then it's already supported.
+        // We do this skip to avoid calling NativeInit(), or creating an extra api to support creating a vendor specific device twice with
+        // memory etc.
+        // It might make more sense if this function was queried with some actual parameters, but as it is now
+        // it is merely a question of "is Vulkan supported at all on this platform" /MAWE
+        return true;
+#else
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan.cpp:1138]: Bug fix for switch

File: engine/graphics/src/vulkan/graphics_vulkan_swap_chain.cpp [hunk 4]
```diff
@@ -316,7 +316,9 @@ namespace dmGraphics
         vkGetPhysicalDeviceSurfaceFormatsKHR(vk_device, surface, &format_count, 0);
         vkGetPhysicalDeviceSurfacePresentModesKHR(vk_device, surface, &present_modes_count, 0);
 
+#if defined(ANDROID)
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan_swap_chain.cpp:319]: Not supported on Switch

File: engine/graphics/src/wscript [hunk 5]
```diff
@@ -128,6 +128,8 @@ def build(bld):
     bld.install_files('${PREFIX}/lib/python', 'waf_graphics.py')
 
     if not waflib.Options.options.skip_build_tests:
+        bld.add_group() # make sure they've built before building the tests
```
Comment (JCash) [inline engine/graphics/src/wscript:131]: Poor fix, but don't feel like doing it proper for now.

File: engine/platform/src/wscript [hunk 6]
```diff
@@ -1,40 +1,55 @@
...
+            platform_source = ['%s/platform_window_native.cpp' % build_util.get_target_os()]
 
-    if platform_glfw_version(bld.env.PLATFORM) == 3:
-        glfw_source = ['platform_window_glfw3.cpp']
-        glfw3_source_vulkan = ['platform_window_glfw3_vulkan.cpp']
+            if build_util.get_target_os() in ('nx64'):
+                lib = bld.stlib(features = 'cxx c ddf',
+                        includes = '.',
+                        target   = 'platform',
+                        source   = bld.path.ant_glob(platform_source))
```
Comment (JCash) [inline engine/platform/src/wscript:20]: I'll merge this to the PS4 branch afterwards, and merge with that code as well

**Review by Jhonnyg — APPROVED** [review]

**Pull request merged**